### PR TITLE
Add events

### PR DESCRIPTION
**Pull request by ahangarha**
*Please remove all italic texts including this one. Feel free to add/remove text as per your need.*

## Description

In this PR I have connected the UI with the logic. Key changes:
- Add logic
- Covert buttons to reusable React components
- Connect UI to the logic

### General requirements

- [x] There are **no linter errors**.
- [x] Followed the **Gitflow**.
- [x] The documentation is **professional**.

### HTML/CSS requirements

- [x] Followed best practices for JavaScript

### Additional notes

There are some faulty behavior which are related to the logic code given to us.

---
Thank you for taking the time to review this pull request.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,34 +1,54 @@
...
+        next: null,
+        operation: null,
+      },
+    };
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onClick(buttonName) {
+    const { calcObj } = this.state;
+    this.setState({
```
Comment (GzimAsani) [inline src/components/Calculator.js]: - [x] When I click one number I should see that number be displayed on the screen. Currently for example, if I click 8 the number does not appear on the screen, it appears only after I click + or -. The end result is being calculated correctly but I think you should work on the user experience. The user should be able to too see the numbers that he/she is clicking

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,34 +1,54 @@
...
-        <div className="btn btn-add btn-operation">+</div>
-        <div className="btn btn-0">0</div>
-        <div className="btn btn-decimal">.</div>
-        <div className="btn btn-equal btn-operation">=</div>
+        <div className="result-section">
+          {Number(calcObj.total)}
+        </div>
+
+        <Button value="AC" onClick={this.onClick} />
+        <Button value="+/-" onClick={this.onClick} />
```
Comment (GzimAsani) [inline src/components/Calculator.js:37]: - [x] When I click 8 and click this button +/- I should see the number `8` changing its value. Currently, I cannot achieve this in your project.

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi @ahangarha               
Good job so far! Your code looks fine and calculations are working as expected, good job :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good job with the gitflow :+1:
- Good PR description :clap: 

## Required Changes ♻️ 
- [ ] _Check comments for more information!_


## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (ahangarha) [reply src/components/Calculator.js]: Thanks. I updated the code and fixed this issue.

File: src/components/Calculator.js (hunk 2 above)
Comment (ahangarha) [reply src/components/Calculator.js:37]: Thanks. I updated the code and fixed this issue.

**Review by julie-ify — APPROVED** [review]
Hi @ahangarha,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# STATUS: APPROVED ✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**